backend/helm/cmd_helm.go

```diff
@@ -0,0 +1,93 @@
+package helm
+
+import (
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	"time"
+)
+
+type ActionConfigInterface interface {
+	getRelease(name string) (*release.Release, error)
+	listReleases(allNamespaces bool) ([]*release.Release, error)
+	uninstallRelease(name string) (*release.UninstallReleaseResponse, error)
+	getReleaseHistory(name string, max int) ([]*release.Release, error)
+	rollbackRelease(name string, version int) error
+}
+
+type ActionConfig struct {
+	config *action.Configuration
+}
+
+func getActionConfig(kubeConfig *rest.Config, namespace string) (*ActionConfig, error) {
+	configFlags := &genericclioptions.ConfigFlags{
+		Namespace: &namespace,
+		WrapConfigFn: func(_ *rest.Config) *rest.Config {
+			return kubeConfig
+		},
+	}
+
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(configFlags, namespace, "secret", func(format string, v ...interface{}) {}); err != nil {
+		return nil, err
+	}
+
+	return &ActionConfig{config: actionConfig}, nil
+}
+
+func (c *ActionConfig) getRelease(name string) (*release.Release, error) {
+	get := action.NewGet(c.config)
+	rel, err := get.Run(name)
+	if err != nil {
+		return nil, err
+	}
+	return rel, nil
+}
+
+func (c *ActionConfig) rollbackRelease(name string, version int) error {
+	rollback := action.NewRollback(c.config)
+	rollback.Version = version
+	rollback.Wait = true
+	rollback.Timeout = 300 * time.Second
```

Comment:
Wydawało mi się ze pisałem o tym ostatnio, ale czy to musi być ustawione na 5 min? Co jeżeli zajmie dłużej dla jakeigoś dużego release'a?

Reply:
Jakaś wartość musi być ustawiona, bo zero nie czeka.